src/hockeypuck/hkp/handler_test.go

```diff
@@ -323,3 +326,16 @@ func (s *HandlerSuite) TestHashQueryResponseUnderLimit(c *gc.C) {
 	// The number of keys should be the same as the number of digests
 	c.Assert(nk, gc.Equals, s.digests)
 }
+
+// Test HashQuery with many duplicate digests
+func (s *HandlerSuite) TestHashQueryReprocess(c *gc.C) {
+	var err error
+	w, req := s.SetupHashQueryTest(c, 100000)
+	c.Assert(err, gc.IsNil)
+	s.handler.HashQuery(w, req, nil)
```

Comment:
I want to mock s.handler.fetchKeysFromDigest() to count how many times it's been called but I haven't decided on the way to do that yet.  @cmars, any suggestions?

Reply:
A few options:

* Rely on mocking the storage layer with a test implementation that does the counting on storage calls instead, or extends the existing test implementation with call counts. This isn't exactly the same thing, but can be close (since `fetchKeysFromDigest` is just making two storage calls). Advantage there is less refactoring, as the storage is already a mockable interface. Another advantage, you can create error scenarios to get more code path coverage inside of `fetchKeysFromDigest`.
* Rename `fetchKeysFromDigest` to something like `defaultFetchKeysFromDigest`. Then declare a Handler field `fetchKeysFromDigest func(digest string) (keys []*openpgp.PrimaryKey, err error)`, initialized to `h.defaultFetchKeysFromDigest`. In tests, you can then override this with your own func that counts (and optionally wraps).

First option is less intrusive, but operates at a layer below. I'd probably go with the first, but either could do what you want.

Reply:
As I've moved the deduplication, this test isn't as important as before. I went with the first one anyway just so we have coverage in that part.